packages/contracts/src/contracts/current/protocol/Exchange/MixinWrapperFunctions.sol

```diff
@@ -26,85 +27,129 @@ contract MixinWrapperFunctions is
-    /// @param v ECDSA signature parameter v.
-    /// @param r ECDSA signature parameters r.
-    /// @param s ECDSA signature parameters s.
+    /// @param signature Maker's signature of the order.
     function fillOrKillOrder(
         address[5] orderAddresses,
-        uint256[6] orderValues,
-        uint256 takerTokenFillAmount,
-        uint8 v,
-        bytes32 r,
-        bytes32 s)
+        uint[6] orderValues,
+        uint takerTokenFillAmount,
+        bytes signature)
         public
     {
         require(fillOrder(
             orderAddresses,
             orderValues,
             takerTokenFillAmount,
-            v,
-            r,
-            s
+            signature
         ) == takerTokenFillAmount);
     }
 
     /// @dev Fills an order with specified parameters and ECDSA signature. Returns false if the transaction would otherwise revert.
     /// @param orderAddresses Array of order's maker, taker, makerToken, takerToken, and feeRecipient.
     /// @param orderValues Array of order's makerTokenAmount, takerTokenAmount, makerFee, takerFee, expirationTimestampInSec, and salt.
     /// @param takerTokenFillAmount Desired amount of takerToken to fill.
-    /// @param v ECDSA signature parameter v.
-    /// @param r ECDSA signature parameters r.
-    /// @param s ECDSA signature parameters s.
+    /// @param signature Maker's signature of the order.
     /// @return Success if the transaction did not revert.
     /// @return Total amount of takerToken filled in trade.
     function fillOrderNoThrow(
         address[5] orderAddresses,
         uint256[6] orderValues,
         uint256 takerTokenFillAmount,
-        uint8 v,
-        bytes32 r,
-        bytes32 s)
+        bytes signature)
         public
         returns (bool success, uint256 takerTokenFilledAmount)
     {
+        // We need to call MExchangeCore.fillOrder using a delegatecall in
+        // assembly so that we can intercept a call that throws. For this, we
+        // need the input encoded in memory in the Ethereum ABI format [1].
+        //
+        // | Offset | Length | Contents                     |
+        // |--------|--------|------------------------------|
+        // | 0      | 4      | function selector            |
+        // | 4      | 160    | address[5] orderAddresses    |
+        // | 164    | 192    | uint256[6] orderValues       |
+        // | 356    | 32     | uint256 takerTokenFillAmount |
+        // | 388    | 32     | len(signature)               |
+        // | 420    | (1)    | signature                    |
+        // | (2)    | (3)    | padding (zero)               |
+        // | (4)    |        | end of input                 |
+        //
+        // (1): len(signature)
+        // (2): 420 + len(signature)
+        // (3): (32 - len(signature)) mod 32
+        // (4): 420 + len(signature) + (32 - len(signature)) mod 32
+        //
+        // [1]: https://solidity.readthedocs.io/en/develop/abi-spec.html
```

Comment:
Looks like this is even more complex: http://solidity.readthedocs.io/en/develop/abi-spec.html#use-of-dynamic-types